viewer/src/logic/commands.ts

```diff
@@ -811,12 +817,46 @@ export function federationButton(command: AvailableCommand<Command.FormFederatio
     })
   );
 
+  const n = locationButtons.length;
+  let index: number = null;
+  const cycle = (update: number) => () => {
+    index = index == null ? 0 : (((index + update) % n) + n) % n;
```

Comment:
Just FYI, since `update` is  +/-1, it can be `(index + update + n) % n`

Reply:
ok, just copied that from the answer to the broken mod in js - hopefully easily recognizable if you've heard of the problem before.